fido/fido.py

```diff
@@ -394,19 +394,19 @@ def identify_file(self, filename):
             # in these cases we try to match the extension instead
             if len(matches) > 0 and self.current_filesize > 0:
                 self.handle_matches(filename, matches, time.clock() - t0, self.matchtype)
-            elif len(matches) == 0 or self.current_filesize == 0:
+            elif extension and (len(matches) == 0 or self.current_filesize == 0):
                 matches = self.match_extensions(filename)
                 self.handle_matches(filename, matches, time.clock() - t0, "extension")
             # only recurse into certain containers, like ZIP or TAR
             container = self.container_type(matches)
             # till here matey!
             if self.zip and self.can_recurse_into_container(container):
-                self.identify_contents(filename, type=container)
+                self.identify_contents(filename, type=container, extension=extension)
         except IOError:
             # print >> sys.stderr, "FIDO: Error in identify_file: Path is {0}".format(filename)
             sys.stderr.write("FIDO: Error in identify_file: Path is {0}\n".format(filename))
 
-    def identify_contents(self, filename, fileobj=None, type=False):
+    def identify_contents(self, filename, fileobj=None, type=False, extension=True):
```

Comment:
What's the point of passing allowing an `extension` arg if it's never used? I think you need to pass `extension` to `walk_zip` and `walk_tar` here.